1config-ui/src/com/brunobonacci/oneconfig/ui/controller.cljs

```diff
@@ -78,6 +81,56 @@
   (swap! state assoc :1config-version version))
 
 
+(defn GETAsync [url]
+  (let [out (chan)
+        handler #(put! out %1)]
+    (GET url {:handler handler
+              :format          :json
+              :response-format :json
+              :keywords?       true
+              :error-handler handler})
+    out))
+
+(defn create-url-a [item]
+  (let [{:keys [key env version change-num]} item]
+    (gs/format "/configs/keys/%s/envs/%s/versions/%s?change-num=%s"
+               (gs/urlEncode key) (gs/urlEncode env)
+               (gs/urlEncode version) change-num)))
+
+(defn compare-selected-items [compare-items]
+  (go
+    (let [urls (map #(create-url-a %) (get compare-items :items-meta))
+          configs [(<! (GETAsync (first urls)))
+                   (<! (GETAsync (last urls)))]
+          selected-entries (map #(:encoded-value %) configs)]
+      (swap! state #(-> %
+                        (assoc-in [:new-entry] empty-new-entry)
+                        (assoc-in [:selected :entries]  selected-entries)
+                        (assoc-in [:entry-management-button-style] "ui inverted disabled button")
+                        (assoc-in [:client-mode] :compare-entry-mode))))))
+
+(defn row-selected
+  [e item]
+  (let [parent (-> e .-target .-parentElement .-parentElement)]
+    (if (.. e -target -checked)
+      (do
+        (swap! state update-in [:selected :counter] inc)
+        (swap! state update-in [:selected :items-meta]  merge item)
```

Comment:
two `swap!` ??? this should probably be merged into one

Reply:
sorry for that, but I'm experimenting....